Lesson14/fikriye/script.js

```diff
@@ -0,0 +1,470 @@
+/*
+3. Lowercase Email
+   - Define a function `normalizeEmail(email)` that returns a lowercased version of the email.
+   - Log the normalized email to the console.
+   - Example: "USER@Example.COM" -> "user@example.com"
+/* */
+function normalizeEmail(email) {
+  return email.toLowerCase();
+}
+
+const email = "USER@Example.COM";
+const normalizedEmail = normalizeEmail(email);  
+console.log(normalizedEmail);
+
+/*
+6. File Extension Check
+   - Define a function `checkFileExtension(filename)` that checks if the filename
+     ends with ".pdf" using .endsWith().
+   - If it does, log: "This is a PDF file."
+   - Otherwise, log: "Not a PDF file."
+*/
+function checkFileExtension(filename) {
+  if (filename.endsWith(".pdf")) {
+    console.log("Bu bir PDF dosyasıdır.");
+  } else {
+    console.log("Bu bir PDF dosyası değildir.");
+  }
+}
+checkFileExtension("dosya.pdf");
+
+
+/*
+7. Compare Numbers (if-else)
+   - Define a function `compareNumbers(a, b)` that:
+     - Logs "a is bigger" if a > b
+     - Logs "b is bigger" if b > a
+     - Logs "Numbers are equal" if they are the same
+*/
+function compareNumbers(a, b) {
+  if (a > b) {
+    console.log("a is bigger");
+  } else if (b > a) {
+    console.log("b is bigger");
+  } else {
+    console.log("Numbers are equal");
+  }
+}
+compareNumbers(2,9);
+/*
+8. Palindrome Check
+  - Define a function `isPalindrome(str)` that checks if `str` is the same
+    forwards and backwards.
+  - If it is, log: "<str> is a palindrome"
+  - Otherwise, log: "<str> is not a palindrome"
+*/
+function isPalindrome(str) {
+  str = str.replace(/\s+/g, '').toLowerCase();
+  if (str === str.split('').reverse().join('')) {
+    console.log(`<str> bir palindromdur`);
+} else {
+    console.log(`<str> bir palindrom değildir`);
+}
+}
+isPalindrome("Timur");
+/*
+9. String Truncation
+   - Define a function `truncateString(text, maxLength)` that uses slice() to
+     cut the string to `maxLength` characters, then appends "..." if it was too long.
+   - Log the final truncated string.
+*/
+function truncateString(text, maxLength) {
+  if (text.length > maxLength){
+    text = text.slice (0,maxLength);
```

Comment:
Since we want to add ... at the end of the string to indicate that we truncated it, you need to add "..." string at the end of the truncated string like this:
```suggestion
    text = text.slice(0, maxLength) + "...";
```